src/main/java/com/google/devtools/build/lib/remote/RemoteSpawnCache.java

```diff
@@ -142,8 +142,9 @@ public CacheHandle lookup(Spawn spawn, SpawnExecutionContext context)
         try (SilentCloseable c = Profiler.instance().profile("RemoteCache.getCachedActionResult")) {
           result = remoteCache.getCachedActionResult(actionKey);
         }
-        if (result != null) {
+        if (result != null && result.getExitCode() == 0) {
           // We don't cache failed actions, so we know the outputs exist.
```

Comment:
So "we don't cache failed actions" is no longer correct for Bazel? If so, this comment is no longer true (even if Bazel won't cache failed actions itself).